### PR TITLE
travis-ci: Enabled debug for linux with gcc

### PR DESCRIPTION
**Pull request by shantonu**
Added --enable-debug in the configure line when building on linux with gcc

Signed-off-by: Shantonu Hossain <shantonu.hossain@intel.com>

### CODE REVIEW

**Comment by pmmccorm** [comment]
Just curious also: what does this gain us?

**Comment by shefty** [comment]
We have an actual bug that the debug build does not work.  This attempts to catch that issue in the future.  The is a sanity check, so we picked a single OS and compiler to test on, rather than doubling the entire set of tests (from 4 to 8).

File: .travis.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ addons:
                         - rpm
 install:
         - ./autogen.sh
-        - ./configure --prefix=$HOME
+        - if [ "$TRAVIS_OS_NAME" = "linux" and "$CC" = "gcc"]; then ./configure --prefix=$HOME --enable-debug; else ./configure --prefix=$HOME; fi
```
Comment (shefty) [inline .travis.yml]: I think we want the enable-debug test to be an additional test, rather than replacing one of the current ones.

Comment (shantonu) [inline .travis.yml]: So you mean, execute the steps as before and an additional step to build with -enable-debug in for linux/gcc?

Comment (shefty) [inline .travis.yml]: > So you mean, execute the steps as before and an additional step to build
> with -enable-debug in for linux/gcc?

Yes

**Comment by shantonu** [comment]
Updated --enable-debug option as a separate test. @shefty, please check if it looks alright.

**Comment by shefty** [comment]
thanks!

**Pull request merged**